sonar/modules/documents/api.py

```diff
@@ -138,6 +143,118 @@ def get_record_by_identifier(cls, identifiers):
 
         return None
 
+    def add_file(self, data, key, **kwargs):
+        """Create file and add it to record.
+
+        kwargs may contain some additional data such as: file label, file type,
+        order and url.
+
+        :param data: Binary data of file
+        :param str key: File key
+        :returns: File object created.
+        """
+        if not kwargs.get('type'):
+            kwargs['type'] = 'file'
+
+        if not kwargs.get('order'):
+            kwargs['order'] = 1
+
+        if not kwargs.get('label'):
+            kwargs['label'] = key
+
+        added_file = super(DocumentRecord, self).add_file(data, key, **kwargs)
+
+        if not added_file:
+            return None
+
+        self.create_fulltext_file(self.files[key])
+        self.create_thumbnail(self.files[key])
+
+        return self.files[key]
+
+    def sync_files(self, file, deleted=False):
+        """Sync files between bucket and records.
+
+        This operation is necessary to make files available in record detail
+        and be indexed.
+
+        :param file: File object.
+        :param deleted: Wether the given file has been deleted or not.
+        """
+        # For documents, a thumbnail and a fulltext file is generated.
+        if not deleted:
+            self.create_fulltext_file(self.files[file.key])
+            self.create_thumbnail(self.files[file.key])
+
+        super(DocumentRecord, self).sync_files(file, deleted)
+
+    def create_fulltext_file(self, file):
+        """Create fulltext file corresponding to give file object.
+
+        :param file: File object.
+        """
+        # If extract fulltext is disabled or file is not a PDF
+        if not current_app.config.get(
+                'SONAR_DOCUMENTS_EXTRACT_FULLTEXT_ON_IMPORT'
+        ) or file.mimetype != 'application/pdf':
```

Comment:
could be interesting (in next PR/US) to extract fulltext from other mimetype (rtf, odt, ...)
In our repository we also have some 'application/pdf+a' mimeType.
Maybe interesting to have a general config parameter to list all mimeTypes available for fulltext extraction.

Reply:
Yes, for sure it could be interesting. I'll will notify that to Miguel.